File: src/main/java/com/github/nsc/de/shake/parser/node/expression/PowNode.java
package com.github.nsc.de.shake.parser.node.expression;

import com.github.nsc.de.shake.lexer.characterinput.position.PositionMap;
import com.github.nsc.de.shake.parser.node.ValuedNode;

public class PowNode extends ExpressionNode {
    public PowNode(PositionMap map, ValuedNode left, ValuedNode right, int operatorPosition) {
        super(map, left, right, operatorPosition);
    }

    @Override
    public char getOperator() {
        return '^';
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/expression/PowNode.kt
package com.github.nsc.de.shake.parser.node.expression

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class PowNode(map: PositionMap, left: ValuedNode, right: ValuedNode, operatorPosition: Int) :
    ExpressionNode(map, left, right, operatorPosition) {
    override val operator: Char
        get() = '^'
}